deal_finder/utils/queue.py

```diff
@@ -0,0 +1,85 @@
+from typing import Any
+
+
+class CircularQueue:
+    def __init__(self, size: int = 50):
+        self._size: int = size
+        self._queue: list = [None for i in range(size)] 
+        self._front: int = -1
+        self._rear: int = -1
+ 
+    def _increment(self, val: int):
+        if not isinstance(val, int):
+            raise TypeError(f"Incorrect type for \"val\": {type(val)}")
+
+        return (val + 1) % self._size 
+
+    def is_full(self) -> bool:
+        return (self._rear + 1) % self._size == self._front
+
+    def is_empty(self) -> bool:
+        return self._front == -1
+
+    def __delete_at_index(self, index: int) -> None:
```

Comment:
Note to self: `index` should be checked for out of bounds condition.\
(Since it is an internal function and we are sure that we are passing down the correct index, we can leave it for now)